Add recipe for flymake-hadolint

buzztaiki opened the pull request:
### Brief summary of what the package does

Flymake backend for hadolint, a Dockerfile linter.

### Direct link to the package repository

https://github.com/buzztaiki/flymake-hadolint

### Your association with the package

Author.

### Relevant communications with the upstream package maintainer

**None needed** 

### Checklist

<!-- Please confirm with `x`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thank you for this!  Very quick first pass --

`flymake-hadolint.el` with checkdoc 0.6.2:
```
flymake-hadolint.el:71: All variables and subroutines might as well have a documentation string
flymake-hadolint.el:81: All variables and subroutines might as well have a documentation string
```

Package and license:
- Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
  See: https://github.com/licensee/licensee

buzztaiki commented:
Thank you for reviewing!

> `flymake-hadolint.el` with checkdoc 0.6.2:

Fixed at https://github.com/buzztaiki/flymake-hadolint/commit/b4dcad70d2894225f65866b5ab07fdda2df2d706.
Sorry, my configuration had `checkdoc-force-docstrings-flag` set to `nil`.


> * Add a LICENSE file that GitHub can detect (e.g. no markup) if possible
>   See: https://github.com/licensee/licensee


Fixed at https://github.com/buzztaiki/flymake-hadolint/commit/9b650a4962d92003708f956a59331dc4dfb348ab.

riscy commented:
> Sorry, my configuration had checkdoc-force-docstrings-flag set to nil.

That's fine - I've actually been thinking of opening an issue to discuss what `checkdoc` links we *really* ought to enforce.

Code look great, and I'm a fan of hadolint so happy to see this package.  Thank you!
